packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+        createdAt: BigInt(TEST_EXECUTION_DATE.getTime()),
+        consumerDocuments: [],
+        stamps: {},
+      };
+
+      expect(actualCreatedAgreementV1).toMatchObject(
+        expectedCreatedAgreementV1
+      );
+
+      const actualAgreementDocumentAddedV1 = await getExpectedAgreementEventV1(
+        newAgreementId,
+        1,
+        {
+          type: "AgreementConsumerDocumentAdded",
+          stream_id: newAgreementId,
+          version: "1",
+        },
+        decodeAgreementDocumentV1FromEvent
+      );
+
+      expect(actualAgreementDocumentAddedV1).toMatchObject({});
+    });
+
+    it("should throw error tenant does not exist", async () => {
+      const authData = getRandomAuthData();
+      const agreementId = generateId<AgreementId>();
+      await expect(
+        agreementService.upgradeAgreement(agreementId, authData, uuidv4())
+      ).rejects.toThrowError(tenantIdNotFound(authData.organizationId));
+    });
+
+    it("should throw error agreement does not exist", async () => {
+      const authData = getRandomAuthData();
+      const agreementId = generateId<AgreementId>();
+
+      const tenantId = authData.organizationId.toString();
+      const tenant = getMockTenant(unsafeBrandId<TenantId>(tenantId));
+
+      await addOneTenant(tenant, tenants);
+      await expect(
+        agreementService.upgradeAgreement(agreementId, authData, uuidv4())
+      ).rejects.toThrowError(agreementNotFound(agreementId));
+    });
+
+    it("should throw error if requester is not consumer", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.ADMIN_ROLE],
+      };
+
+      const tenantId = unsafeBrandId<TenantId>(authData.organizationId);
+      const tenant = getMockTenant(tenantId);
+      await addOneTenant(tenant, tenants);
+
+      const agreementId = generateId<AgreementId>();
+      const agreement = {
+        ...getMockAgreement(
+          generateId<EServiceId>(),
+          generateId<TenantId>(),
+          agreementState.active
+        ),
+        id: agreementId,
+      };
+
+      await addOneAgreement(agreement, postgresDB, agreements);
+      await expect(
+        agreementService.upgradeAgreement(agreementId, authData, uuidv4())
+      ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
+    });
+
+    it("should throw error if agreement is not in upgradable states", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.ADMIN_ROLE],
+      };
+
+      const tenantId = unsafeBrandId<TenantId>(authData.organizationId);
+      const tenant = getMockTenant(tenantId);
+      await addOneTenant(tenant, tenants);
+
+      const agreementId = generateId<AgreementId>();
+      const agreement = {
+        ...getMockAgreement(
+          generateId<EServiceId>(),
+          tenantId,
+          agreementState.active
+        ),
+        id: agreementId,
+        state: agreementState.rejected,
```

Comment:
If you use the following you can increase coverage testing for all the not-expected states:

```ts
Object.values(agreementState).filter(
            (s) => !agreementUpgradableStates.includes(s)
          )
```

Reply:
✅ Done with [commit d35f](https://github.com/pagopa/interop-be-monorepo/pull/392/commits/d35fd1d520fa27a2bcffa724ee09266ebba0cbdc)